### PR TITLE
Added Guide Settings Panel

### PR DESCRIPTION
**Pull request by insominx**
Added the ability to adjust stencil/guide related values from a new Guide Settings panel.  This depends on #158 as that PR laid the groundwork for the shader adjustments to work correctly.

The new panel allows tweaking:
- pointer snap distance (moved from the guide panel to this new one)
- grid size (how much space is between the grid lines)
- grid line width
- frame width (how thick the "outer" edge lines are)

https://youtu.be/9EM-pqHlFDs

### CODE REVIEW

File: Assets/Scripts/GUI/BasePanel.cs [hunk 1]
```diff
@@ -103,7 +103,8 @@ public enum PanelType
             LabsMobile,
             ReferenceMobile,
             CameraPath,
-            BrushLab
+            BrushLab,
+            StencilSettings = 20200
```
Comment (andybak) [inline Assets/Scripts/GUI/BasePanel.cs:107]: Yay! Someone actually read my docs!

**Comment by andybak** [comment]
I guess my only concern is whether this is too much detail to be exposing in the UI. Is there a danger of drowning the user in potential tweakable settings? Especially mainly cosmetic things.

**Comment by insominx** [comment]
> I guess my only concern is whether this is too much detail to be exposing in the UI. 
The alternative is in ``UserConfig`` but that doesn't allow fine tuning very easily which is what this requires.

> Is there a danger of drowning the user in potential tweakable settings? Especially mainly cosmetic things. I was probably getting a bit uneasy with grid line width and especially frame width.
>
If the settings were required to be manipulated, "drowning" might be more applicable.  In this case, the defaults are sensible and the settings are there for those that care.  What might not be obvious is that some of those settings allow you to accomplish oft desired (at least by me) cosmetics:

- controlling frame width allows us to remove the frame entirely
- controlling line width allows us to remove the lines entirely
- controlling grid size also allows us to make the surface completely covered by one color

> I'd be happy with this commit but there's a potential "death by 1000 cuts" situation here and at some point we might have to draw a line.

Isn't this why we have beginner and advanced modes?  If you're in advanced (where this panel lives) then you're not too afraid of some additional complexity.

**Comment by andybak** [comment]
> Isn't this why we have beginner and advanced modes? If you're in advanced (where this panel lives) then you're not too afraid of some additional complexity.

Yeah although I think there's still potentially a "too many options" ceiling even for advanced users. But this is a "keep an eye on for the future" comment. We're nowhere close to the limit at the moment.

Plus you just gave me some use cases I hadn't even thought of.

**Review by mikeskydev — APPROVED** [review]
LGTM! @andybak?

**Review by andybak — APPROVED** [review]
Yep. Looks great and is really useful Two small quibbles:

1. The tooltip for the plane guide still says "cube guide"
2. The button to open the settings probably should be a smaller button or icon rather than the same size as the stencil icons. Can't think where in the UI I've seen this but it feels like it would be better that way.

1 is already a bug in the release so we can fix it after merging
2 is open for debate - so I vote that we merge anyway for now.

**Pull request merged**